spec/02-integration/18-hybrid_rpc/08-sync_v2_get_delta_spec.lua

```diff
@@ -0,0 +1,78 @@
+local helpers = require "spec.helpers"
+
+
+-- register a rpc connected event in custom plugin rpc-get-delta-test
+-- ENABLE rpc sync on cp side for testing sync.v2.get_delta
+-- DISABLE rpc sync on dp side
+for _, strategy in helpers.each_strategy() do
+  describe("Hybrid Mode RPC #" .. strategy, function()
+
+    lazy_setup(function()
+      helpers.get_db_utils(strategy, {
+        "clustering_data_planes",
+      }) -- runs migrations
+
+      assert(helpers.start_kong({
+        role = "control_plane",
+        cluster_cert = "spec/fixtures/kong_clustering.crt",
+        cluster_cert_key = "spec/fixtures/kong_clustering.key",
+        database = strategy,
+        cluster_listen = "127.0.0.1:9005",
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+        plugins = "bundled",
+        nginx_worker_processes = 4, -- multiple workers
+        cluster_rpc = "on", -- enable rpc
+        cluster_rpc_sync = "on", -- enable rpc sync
+      }))
+
+      assert(helpers.start_kong({
+        role = "data_plane",
+        database = "off",
+        prefix = "servroot2",
+        cluster_cert = "spec/fixtures/kong_clustering.crt",
+        cluster_cert_key = "spec/fixtures/kong_clustering.key",
+        cluster_control_plane = "127.0.0.1:9005",
+        proxy_listen = "0.0.0.0:9002",
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+        plugins = "bundled,rpc-get-delta-test",
+        nginx_worker_processes = 4, -- multiple workers
+        cluster_rpc = "on", -- enable rpc
+        cluster_rpc_sync = "off", -- disable rpc sync
+      }))
+    end)
+
+    lazy_teardown(function()
+      helpers.stop_kong("servroot2")
+      helpers.stop_kong()
+    end)
+
+    describe("sync.v2.get_delta works", function()
+      it("in cp side", function()
+        local name = "servroot2/logs/error.log"
+
+        -- dp logs
+        helpers.pwait_until(function()
+          assert.logfile(name).has.line(
```

Comment:
This kind of assertion already has timeout as the input parameter, so the `pwait_until` is not required.

https://github.com/Kong/kong/blob/e9510e9abde5400659e7c57eb2304f5ebb8fd0be/spec/03-plugins/04-file-log/01-log_spec.lua#L562

Reply:
Not sure about it, if assertion fails `wait_until` may crush, so `pwait_until` is a better choice.

Reply:
I checked the code in spec/internal/wait.lua, pwait_until is different from wait_until, it can deal with assertion failings.

Reply:
Got, let me try it.

Reply:
It works, but the tests take more time than pwait_until, so my opinion is keeping it.

Reply:
How much extra time is taken by assertion with timeout roughly?

Reply:
pwait_until: 10.874588 seconds
line with timeout: 43.034279 seconds

Reply:
changed.